Modules/gcmodule.c

```diff
@@ -1953,6 +1957,48 @@ gc_get_freeze_count_impl(PyObject *module)
 }
 
 
+static int
+immortalize_object(PyObject *obj, PyGC_Head *permanent_gen)
+{
+    if (_Py_IsImmortal(obj)) {
+      return 0;
+    }
+
+    _Py_SetImmortal(obj);
+    /* Special case for PyCodeObjects since they don't have a tp_traverse */
+    if (PyCode_Check(obj)) {
+        PyCodeObject *code = (PyCodeObject *)obj;
+        _Py_SetImmortal(code->co_code);
+        _Py_SetImmortal(code->co_consts);
+        _Py_SetImmortal(code->co_names);
+        _Py_SetImmortal(code->co_varnames);
+        _Py_SetImmortal(code->co_freevars);
+        _Py_SetImmortal(code->co_cellvars);
```

Comment:
Make contents of these tuples immortal too.
Especially, co_consts is important, because LOAD_CONST will INCREF.